src/coreclr/jit/hwintrinsicxarch.cpp

```diff
@@ -1284,6 +1284,43 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
             break;
         }
 
+        // TODO-XARCH-AVX512 see how to merge this into below
+        case NI_Vector512_Equals:
+        {
+            assert(sig->numArgs == 2);
+
+            if (IsBaselineVector512IsaSupported())
```

Comment:
These should be assertable, we shouldn't get this far if the ISA isn't supported.

Reply:
(Make sure you use the DebugOnly variant `assert(IsBaselineVector512IsaSupportedDebugOnly());`)

Reply:
Done